### ISSUE TITLE
App not able to connect to bluetooth module?

### ISSUE THREAD

**Issue by MuckYu** [leonardok/ascale#1]
Hi, I imported the android folder into android studio and build an apk, which I installed on my phone.

The app opens fine and loads the "Find scale" screen.

However no bluetooth module is showing up. (I tested both, HC-05 and HC-06)
I can connect to them via other app.

Am I missing something on how to properly connect it with the app?

**Comment by leonardok** [comment]
Does the arduino looks ok? Can you see the bluetooth in the android bluetooth scanner?

**Comment by MuckYu** [comment]
Yes the arduino and the HC-05 module are both working fine. I am able to find it in the android bluetooth list. 

I also tried several HC-05 arduino related apps on the playstore and they connect to it.

Maybe I am doing something wrong with the .apk export? Any settings I should be aware of?

I am using android 6.0.1

**Comment by MuckYu** [comment]
Yes the bluetooth module is showing up in the BT list on my android. I can
also connect to it via several HC-05 arduino apps from the playstore.

But in this app the list is empty.

Maybe I did something wrong when I exported it to an .apk? Any special
settings I should be aware of?

**Comment by MuckYu** [comment]
Using android 6.0.1

On 30 Aug 2017 11:20 a.m., wrote:

Yes the bluetooth module is showing up in the BT list on my android. I can
also connect to it via several HC-05 arduino apps from the playstore.

But in this app the list is empty.

Maybe I did something wrong when I exported it to an .apk? Any special
settings I should be aware of?

On 30 Aug 2017 7:23 a.m., "Leonardo Korndörfer" <notifications@github.com>
wrote:

Does the arduino looks ok? Can you see the bluetooth in the android
bluetooth scanner?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/leonardok/ascale/issues/1#issuecomment-325832490>, or mute
the thread

**Comment by leonardok** [comment]
The list has a pulldown refresh. Did you try to pull it down to refresh? It is obvious but I remember now that I haven't put any kind of visual indication that the blueooth devices list has pulldown refresh.
Can you connect your phone to Android Studio and check the console output for any errors?

It doesn't looks like you are doing something wrong...

In `FindScaleFragment.java`, in the `refreshDeviceList` method there is a debug line for when it finds a device.

**Comment by MuckYu** [comment]
Yes, I tried the pull down menu already.

I have now connected my phone to the android studio console. 

Surprisingly in the android monitor tab I see this line:

> 08-31 01:03:32.704 21783-29221/me.leok.scaleduino V/FindScaleFragment: Found device: HC-05
08-31 01:03:32.706 21783-29221/me.leok.scaleduino V/FindScaleFragment: Ended device scan

But on the phone it's empty.
Any ideas?

**Comment by leonardok** [comment]
Oh! Probably I haven't updated my arduino code...
The problem is that there is a condition in the android code to show only devices with "ascale" as name. Do you see in lines 155 though 157:

```
if (scaleBluetoothName.toString().equals(device.getName())) {
    scaleDevices.add(device);
}
```
`scaleBluetoothName` is defined to "ascale". Can you remove that condition so `scaleDevices.add(device);` is called to all BT devices?

Change is basically:

```
if (scaleBluetoothName.toString().equals(device.getName())) {
    scaleDevices.add(device);
}
```

becomes:

```
scaleDevices.add(device);
```

**Comment by MuckYu** [comment]
That worked, thanks a lot!

**Comment by leonardok** [comment]
Sure! I will leave this open so I can change the code later.
Any other thing just drop me a message!

**Comment by MuckYu** [comment]
Okay I got another problem I think.

Now I am connected to the bluetooth module, but I don't get any value on the app from the scale.

In the arduino console I get the data just fine. Is there still something I have to adjust in the code?
